src/Illuminate/Http/Request.php

```diff
@@ -812,4 +819,23 @@ public function __get($key)
     {
         return Arr::get($this->all(), $key, fn () => $this->route($key));
     }
+
+    /**
+     * {@inheritdoc}
+     *
+     * Override to clear cache when Accept header changes.
+     */
+    #[\Override]
+    public function getAcceptableContentTypes(): array
+    {
+        $currentAcceptHeader = $this->headers->get('Accept');
+
+        if ($this->cachedAcceptHeader !== $currentAcceptHeader) {
+            $this->acceptableContentTypes = null;
+        }
+
+        $this->cachedAcceptHeader = $currentAcceptHeader;
```

Comment:
I think this statement can be moved into the `if` block so the assignment only happens when needed.